plugins/woocommerce-blocks/assets/js/blocks/product-filter/inner-blocks/price-filter/frontend.ts

```diff
@@ -76,55 +76,168 @@ const debounceUpdate = debounce( ( context, element, event ) => {
 	);
 }, 1000 );
 
-store< PriceFilterStore >( 'woocommerce/product-filter-price', {
-	state: {
-		rangeStyle: () => {
-			const { minPrice, maxPrice, minRange, maxRange } =
-				getContext< PriceFilterContext >();
-
-			return [
-				`--low: ${
-					( 100 * ( minPrice - minRange ) ) / ( maxRange - minRange )
-				}%`,
-				`--high: ${
-					( 100 * ( maxPrice - minRange ) ) / ( maxRange - minRange )
-				}%`,
-			].join( ';' );
-		},
-		formattedMinPrice: () => {
-			const { minPrice } = getContext< PriceFilterContext >();
-			return formatPrice( minPrice, getCurrency( { minorUnit: 0 } ) );
-		},
-		formattedMaxPrice: () => {
-			const { maxPrice } = getContext< PriceFilterContext >();
-			return formatPrice( maxPrice, getCurrency( { minorUnit: 0 } ) );
-		},
-	},
-	actions: {
-		updateProducts: ( event: HTMLElementEvent< HTMLInputElement > ) => {
-			const context = getContext< PriceFilterContext >();
+const constrainRangeSliderValues = (
+	/**
+	 * Tuple containing min and max values.
+	 */
+	values: [ number, number ],
```

Comment:
What is the benefit of using a tuple here over passing two parameters? We have to prepare the array to pass into this function, and then destructure it inside the function.

Reply:
So that is just a utility function I borrowed from https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/assets/js/base/components/price-slider/constrain-range-slider-values.ts and since filter will eventually replace the legacy, I copied that over.